### PR TITLE
[BEAM-1645] Populate display data on Window.Assign

### PR DESCRIPTION
**Pull request by kennknowles**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by kennknowles** [comment]
@tgroh @bjchambers I haven't actually run this yet and looked at the UI, but you might be interested.

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/Window.java [hunk 1]
```diff
@@ -487,38 +487,14 @@ private boolean canProduceMultiplePanes(WindowingStrategy<?, ?> strategy) {
...
-          .withLabel("Window Closing Behavior"));
-      }
-
-      if (outputTimeFn != null) {
-        builder.add(DisplayData.item("outputTimeFn", outputTimeFn.getClass())
-          .withLabel("Output Time Function"));
-      }
+      populateWindowingStrategyDisplayData(
+          WindowingStrategy.of(windowFn)
+              .withAllowedLateness(allowedLateness)
```
Comment (kennknowles) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/Window.java]: Here I need to build the windowing strategy conditionally.

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/Window.java [hunk 2]
```diff
@@ -557,6 +533,52 @@ private Assign(WindowingStrategy updatedStrategy) {
...
+    public void populateDisplayData(DisplayData.Builder builder) {
+      super.populateDisplayData(builder);
+      populateWindowingStrategyDisplayData(updatedStrategy, builder);
+    }
+  }
+
+  private static void populateWindowingStrategyDisplayData(
+      WindowingStrategy<?, ?> windowingStrategy, DisplayData.Builder builder) {
+
+    builder
```
Comment (kennknowles) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/Window.java]: The windowFn and closing behavior are not conditionalized the same way. Cleaner would be to explicitly represent a diff vs a completed strategy.

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/Window.java (hunk 1 above)
Comment (tgroh) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/Window.java]: I feel as though allowing a null `WindowFn` should not be permitted by `WindowingStrategy`, as an aside.

Comment (tgroh) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/Window.java]: We do lack the context here to construct the entire windowing strategy, which is unfortunate. `Window.Assign` is provided it on construction, but I don't know if there's any way to pass it to the identity flatten's display data.

**Comment by tgroh** [comment]
This is probably preempted by https://github.com/apache/beam/pull/2258

**Comment by tgroh** [comment]
@kennknowles is this still worthwhile?

**Comment by kennknowles** [comment]
I was going to ask you the same thing. I think it was addressed adequately while I was away.

**Pull request closed without merging**